lib/node_modules/@stdlib/blas/base/sgemv/lib/sgemv.js

```diff
@@ -0,0 +1,223 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable */ // FIXME
+
+'use strict';
+
+// MODULES //
+
+var xerbla = require( '@stdlib/blas/base/xerbla' );
+var max = require( '@stdlib/math/base/special/max' );
+
+// MAIN //
+
+/**
+* Performs one of the matrix-vector operations `y = alpha*A*x + beta*y` or `y = alpha*A**T*x + beta*y`.
+*
+* @param {string} order - storage layout
+* @param {string} trans - specifies the operation to be performed
+* @param {NonNegativeInteger} M - number of rows in the matrix `A`
+* @param {NonNegativeInteger} N - number of columns in the matrix `A`
+* @param {number} alpha - scalar constant
+* @param {Float32Array} A - matrix of coefficients
+* @param {PositiveInteger} LDA - stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`)
+* @param {Float32Array} x - an `M` element vector
+* @param {integer} strideX - `x` stride length
+* @param {number} beta - scalar constant
+* @param {Float32Array} y - an `N` element vector
+* @param {integer} strideY - `y` stride length
+* @returns {Float32Array} `y`
+*
+* @example
+* var Float32Array = require( '@stdlib/array/float64' );
+*
+* var trans = 'N';
+*
+* var M = 2;
+* var N = 3;
+* var lda = 3;
+*
+* var alpha = 1.0;
+* var beta = 1.0;
+*
+* var A = new Float32Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+* var x  = new Float32Array( [ 1.0, 1.0, 1.0 ] );
+* var y = new Float32Array( [ 1.0, 1.0 ] );
+*
+* sgemv( 'row-major', trans, M, N, alpha, A, lda, x, 1, beta, y, 1 );
+* // y => <Float32Array>[ 7.0, 16.0 ]
+*/
+function sgemv( order, trans, M, N, alpha, A, LDA, x, strideX, beta, y, strideY ) {
+	var info;
+	var lenx;
+	var leny;
+	var temp;
+	var ix;
+	var iy;
+	var jx;
+	var jy;
+	var kx;
+	var ky;
+	var i;
+	var j;
+
+	info = 0;
+	if ( ( trans.toUpperCase() !== 'N' ) && ( trans.toUpperCase() !== 'T' ) && ( trans.toUpperCase() !== 'C' ) ) {
+		info = 1;
+	} else if( M < 0 ) {
+		info = 2;
+	} else if ( N < 0 ) {
+		info = 3;
+	} else if ( LDA < max( 1, M ) ) {
+		info = 6;
+	} else if ( strideX === 0 ) {
+		info = 8;
+	} else if ( strideY === 0 ) {
+		info = 11;
+	}
+	if ( info !== 0 ) {
+		xerbla( 'sgemv', info );
```

Comment:
We want to explicitly raise an exception here. We don't have a JS `xerbla` implementation, as it is not needed.

Reply:
If we use the Js throw error statement, we won't be able to return as done in the Fortran code, as the code will break and throw an exception. As a result, we cannot include a case in the test suite for which M or N is less than 0.

Reply:
`xerbla` doesn't behave as you think it does (see https://www.netlib.org/lapack/explore-html/d9/d64/_s_r_c_2xerbla_8f_source.html). The key statement is `stop`. In Fortran, that will stop execution after printing the error message. That is effectively the same thing as throwing an exception in JS. In which case, returning the output array is not what we want to do.

You can also see that, in GSL, they abort: https://git.savannah.gnu.org/cgit/gsl.git/tree/cblas/xerbla.c.